Redo: Quickstart, Install Epinio, Install on RKE2 and System requirements (#242)
epinio/docs: docs/tutorials/quickstart.md

---
sidebar_label: "Quickstart"
sidebar_position: 1
title: ""
---

# Quickstart

> This guide will help you deploy and use Epinio with default options suitable for evaluation or testing purposes on existing Kubernetes cluster. For advanced Epinio deployment scenarios look into [Installation Section](../installation/install_epinio.md).

Reviewer: ```suggestion
This guide will help you deploy and use Epinio with default options suitable for evaluation or testing purposes on an existing Kubernetes cluster. For advanced Epinio deployment scenarios look into [Installation Section](../installation/install_epinio.md).
```

## Installation

> Make sure your Kubernetes environment fulfills [Epinio Requirements](../references/system_requirements/global.md). A **default StorageClass**  as well as a **default IngressClass** are required. If you do not have a suitable Kubernetes cluster yet, you can follow the [RKE2 Installation](../howtos/install_epinio_on_rke.md) section.

Reviewer: ```suggestion
Make sure your Kubernetes environment fulfills the [Epinio Requirements](../references/system_requirements/global.md). A **default StorageClass**  as well as a **default IngressClass** are required. If you do not have a suitable Kubernetes cluster yet, you can follow the [RKE2 Installation](../howtos/install_epinio_on_rke.md) section.
```

### Deploy Epinio

> Get the `INTERNAL-IP` value of the first Kubernetes node with `kubectl get nodes -o wide` command. Later you will use this value along with a wildcard DNS service domain (for eg. sslip.io) as helm `global.domain` value for installing Epinio.

Reviewer: ```suggestion
Run the  `kubectl get nodes -o wide` command to get the `INTERNAL-IP` value of the first Kubernetes node. Later you will use this value along with a wildcard DNS service domain (for eg. `sslip.io`) as helm `global.domain` value for installing Epinio.
```

:::tip
If you use Local Kubernetes Cluster, the value should be `127.0.0.1` regardless of the output from the command above. Please refer to your Local Kubernetes Cluster documentation for the IP address details of the Ingress endpoint.
:::

#### Install cert-manager
```bash
helm repo add jetstack https://charts.jetstack.io
helm repo update
helm upgrade --install cert-manager jetstack/cert-manager \
    --namespace cert-manager --create-namespace \
    --set installCRDs=true
```

#### Install Epinio

> The Epinio can be deployed by using `helm` as follows, replace the `<INTERNAL-IP>` placeholder:

Reviewer: ```suggestion
Then Epinio can be deployed by using `helm` as shown below. Replace the `<INTERNAL-IP>` placeholder with the correct IP address:
```

```bash
helm repo add epinio https://epinio.github.io/helm-charts
helm repo update
helm upgrade --install epinio epinio/epinio \
    --namespace epinio --create-namespace \
    --set global.domain=<INTERNAL-IP>.sslip.io
```

> You can point your browser and/or epinio CLI to `https://epinio.<INTERNAL-IP>.sslip.io` address then.

Reviewer: ```suggestion
You can then point your browser and/or epinio CLI to the `https://epinio.<INTERNAL-IP>.sslip.io` url.
```

#### Download the Epinio CLI binary

> Download the Epinio CLI for the desired version and architecture from the Assets section of https://github.com/epinio/epinio/releases/.

Author: resolved by a59a0e187a304ab86d4f9adb7069be342efe0cb6

---
sidebar_label: "Quickstart"
sidebar_position: 1
title: ""
---

# Quickstart
This guide will help you deploy and use Epinio with default options suitable for evaluation or testing purposes on an existing Kubernetes cluster. For advanced Epinio deployment scenarios look into [Installation Section](../installation/install_epinio.md).

## Installation
Make sure your Kubernetes environment fulfills the [Epinio Requirements](../references/system_requirements/global.md). A **default StorageClass**  as well as a **default IngressClass** are required. If you do not have a suitable Kubernetes cluster yet, you can follow the [RKE2 Installation](../howtos/install_epinio_on_rke.md) section.

### Deploy Epinio
Run the  `kubectl get nodes -o wide` command to get the `INTERNAL-IP` value of the first Kubernetes node. Later you will use this value along with a wildcard DNS service domain (for eg. `sslip.io`) as helm `global.domain` value for installing Epinio.

:::tip

> If you use Local Kubernetes Cluster, the value should be `127.0.0.1` regardless of the output from the command above. Please refer to your Local Kubernetes Cluster documentation for the IP address details of the Ingress endpoint.

Author: k3d is a bit specific in sense the k8s is usually running in docker on host and the networking works differently (using host network?) - RD, DD and so on are running in an extra VM and they are reachable over localhost forwarded to the VM IP.

> If you use Local Kubernetes Cluster, the value should be `127.0.0.1` regardless of the output from the command above. Please refer to your Local Kubernetes Cluster documentation for the IP address details of the Ingress endpoint.

Author: Indeed the 127.0.0.1 approach doesn't match with k3d but we cannot cover everything - I consider this discrepancy as acceptable.
